ledger/src/shred/wire.rs

```diff
@@ -34,19 +35,30 @@ fn get_shred_size(shred: &[u8]) -> Option<usize> {
 }
 
 #[inline]
-pub fn get_shred(packet: &Packet) -> Option<&[u8]> {
-    let data = packet.data(..)?;
+pub fn get_shred<'a, P>(packet: P) -> Option<&'a [u8]>
+where
+    P: Into<PacketRef<'a>>,
+{
+    let data = packet.into().data(..)?;
     data.get(..get_shred_size(data)?)
 }
 
 #[inline]
-pub fn get_shred_mut(packet: &mut Packet) -> Option<&mut [u8]> {
-    let buffer = packet.buffer_mut();
-    buffer.get_mut(..get_shred_size(buffer)?)
+pub fn get_shred_mut<'a>(packet: &'a mut PacketRefMut) -> Option<&'a mut [u8]> {
+    // This function is used only in turbine for re-signing shreds.
+    match packet {
```

Comment:
FYI we resign < 64 shreds per block. No big deal if we need to clone for it.

Reply:
Once we get rid of the old `Packet`, we will be able to avoid copying there and just return a sliced `Bytes` :saluting_face: